### ISSUE TITLE
Bug in the clWRF wrfxtrm files?

### ISSUE THREAD

**Issue by dmoreno11** [wrf-model/WRF#1367]
**Describe the bug**
Hello all,

I noticed that the clWRF statistics files (wrfxtrm) have some erroneous values (eg., surface temperatures of thousands of K, among others). This only happens for some time steps (days, for these files) but when it happens it affects all variables contained in the wrfxtrm files.

I have confirmed in the corresponding wrfout files that, for the same variables and times, these erroneous values are not present. 
The first screenshot is from the wrfout file and the second for the corresponding wrfxtrm file, for the surface temeprature. The wrfout file has output every 3h and wrfxtrm is daily.

![good](https://user-images.githubusercontent.com/77486171/104713073-464b3700-571b-11eb-899f-a9abac7b9618.png)

![bad](https://user-images.githubusercontent.com/77486171/104713112-5236f900-571b-11eb-9468-0b3dfddd97d3.png)

Any insights on this?

**Comment by dudhia** [comment]
Two questions. 1) is this adaptive steps? 2) is this the most recent WRF
version.
Fixes have been added for wrfxtrm recently.

**Comment by dmoreno11** [comment]
1) yes, I used adaptative time-step
2) It's WRF 4.2. 

I saw that some bugs in clWRF are present in this WRF tag, but the reported bugs only mentioned the **times** where max and min values were obtained, not the **values** per se.

**Comment by dmoreno11** [comment]
Moreover, the crazy values are present in all vars, even the mean ones, and they occur for all vars at the same timesteps.

**Comment by dudhia** [comment]
Does it affect the whole domain when it happens?
I may suspect that adaptive steps are not fully accounted for with this
time averaging.
Is it only the means that are bad or also the max/min?

**Comment by dmoreno11** [comment]
Yes, when it happens all domain is affected, as it can be seen in the screeenshot.

It happens for the mean, max, min and std for T2, skintemp, RAINC and RAINNC; only mean and std for UV10m, SPDUV10MEAN and Q2 (max and min seem OK)

**Comment by dmoreno11** [comment]
But this only happens for some days in a year, not always or often.

**Comment by dmoreno11** [comment]
I did 20 year-long runs and some years were 100% ok.

**Comment by dudhia** [comment]
So the T2max is also unrealistically large. How large?

**Comment by dudhia** [comment]
I also wonder if a restart re-run shows repeatability for the problem
periods.

**Comment by dmoreno11** [comment]
Too large (thousands of K)... But negative!

Same order of magnitude for all erroneous values, regardless of the variable. Always above one thousand, but while some vars are + thousands, others are - thousands

**Comment by dmoreno11** [comment]
cdo infon out.nc | grep T2MEAN
     5 : 1994-12-08 00:00:00       0    18000       0 :      273.55      285.59      289.35 : T2MEAN        
    41 : 1994-12-09 00:00:00       0    18000       0 :      272.66      285.18      289.46 : T2MEAN        
    77 : 1994-12-10 00:00:00       0    18000       0 :      267.91      281.80      287.02 : T2MEAN        
   113 : 1994-12-11 00:00:00       0    18000       0 :      271.44      282.98      287.18 : T2MEAN        
   149 : 1994-12-12 00:00:00       0    18000       0 :      275.36      285.55      288.12 : T2MEAN        
   185 : 1994-12-13 00:00:00       0    18000       0 :      270.10      284.39      288.96 : T2MEAN        
   221 : 1994-12-14 00:00:00       0    18000       0 :      269.99      282.49      287.26 : T2MEAN        
   257 : 1994-12-15 00:00:00       0    18000       0 :      270.78      282.98      287.01 : T2MEAN        
   293 : 1994-12-16 00:00:00       0    18000       0 :      272.49      283.34      287.51 : T2MEAN        
   329 : 1994-12-17 00:00:00       0    18000       0 :      273.36      284.05      287.80 : T2MEAN        
   365 : 1994-12-18 00:00:00       0    18000       0 :      268.98      282.89      287.78 : T2MEAN        
   401 : 1994-12-19 00:00:00       0    18000       0 :      270.30      282.89      286.85 : T2MEAN        
   437 : 1994-12-20 00:00:00       0    18000       0 :      264.06      282.16      286.97 : T2MEAN        
   473 : 1994-12-21 00:00:00       0    18000       0 :      261.61      281.36      287.19 : T2MEAN        
   509 : 1994-12-22 00:00:00       0    18000       0 :      263.40      281.92      287.69 : T2MEAN        
   545 : 1994-12-23 00:00:00       0    18000       0 :      266.40      281.78      288.04 : T2MEAN        
   581 : 1994-12-24 00:00:00       0    18000       0 :      269.19      281.54      288.23 : T2MEAN        
   617 : 1994-12-25 00:00:00       0    18000       0 :      268.84      282.36      289.28 : T2MEAN        
   653 : 1994-12-26 00:00:00       0    18000       0 :      271.02      284.20      289.82 : T2MEAN        
   689 : 1994-12-27 00:00:00       0    18000       0 :      263.68      281.87      289.27 : T2MEAN        
   725 : 1994-12-28 00:00:00       0    18000       0 :      263.60      281.50      288.26 : T2MEAN        
   761 : 1994-12-29 00:00:00       0    18000       0 :      268.03      281.94      287.07 : T2MEAN        
   **797 : 1994-12-30 00:00:00       0    18000       0 :     -4682.7      18276.      27865. : T2MEAN**        
   833 : 1994-12-31 00:00:00       0    18000       0 :      276.36      286.80      289.82 : T2MEAN        
   869 : 1995-01-01 00:00:00       0    18000       0 :      276.73      287.56      292.33 : T2MEAN        
   905 : 1995-01-02 00:00:00       0    18000       0 :      275.94      286.95      290.30 : T2MEAN        
   941 : 1995-01-03 00:00:00       0    18000       0 :      274.56      286.07      289.53 : T2MEAN        
   977 : 1995-01-04 00:00:00       0    18000       0 :      275.04      286.81      290.56 : T2MEAN        
  1013 : 1995-01-05 00:00:00       0    18000       0 :      272.97      284.45      288.59 : T2MEAN        
  1049 : 1995-01-06 00:00:00       0    18000       0 :      269.66      282.31      286.98 : T2MEAN        
  1085 : 1995-01-07 00:00:00       0    18000       0 :      268.70      283.11      288.34 : T2MEAN

**Comment by dmoreno11** [comment]
So, looking only at this day:

   793 : 1994-12-30 00:00:00       0    18000       0 :     -11.555      7.2105      14.060 : T2MIN         
   794 : 1994-12-30 00:00:00       0    18000       0 :      1.6316      12.685      16.565 : T2MAX         
   795 : 1994-12-30 00:00:00       0    18000       0 :  5.1846e+05  5.2164e+05  5.2272e+05 : TT2MIN        
   796 : 1994-12-30 00:00:00       0    18000       0 :  5.2068e+05  5.2259e+05  5.2272e+05 : TT2MAX        
   797 : 1994-12-30 00:00:00       0    18000       0 :     -4682.7      18276.      27865. : T2MEAN        
   798 : 1994-12-30 00:00:00       0    18000       0 :      2604.6  2.1459e+05  4.2305e+05 : T2STD         
   799 : 1994-12-30 00:00:00       0    18000       0 :   0.0016097   0.0052030   0.0070992 : Q2MIN         
   800 : 1994-12-30 00:00:00       0    18000       0 :   0.0051122   0.0086867    0.010354 : Q2MAX         
   801 : 1994-12-30 00:00:00       0    18000       0 :  5.1755e+05  5.2132e+05  5.2257e+05 : TQ2MIN        
   802 : 1994-12-30 00:00:00       0    18000       0 :  5.2132e+05  5.2266e+05  5.2272e+05 : TQ2MAX        
   803 : 1994-12-30 00:00:00       0    18000       0 :      6.7547      12.742      16.212 : Q2MEAN        
   804 : 1994-12-30 00:00:00       0    18000       0 :    0.026496    0.093160     0.14528 : Q2STD         
   805 : 1994-12-30 00:00:00       0    18000       0 :     -17.812      8.3538      16.390 : SKINTEMPMIN   
   806 : 1994-12-30 00:00:00       0    18000       0 :     0.23215      12.653      21.942 : SKINTEMPMAX   
   807 : 1994-12-30 00:00:00       0    18000       0 :  5.1998e+05  5.2153e+05  5.2239e+05 : TSKINTEMPMIN  
   808 : 1994-12-30 00:00:00       0    18000       0 :  5.2128e+05  5.2243e+05  5.2272e+05 : TSKINTEMPMAX  
   809 : 1994-12-30 00:00:00       0    18000       0 :     -5981.3      19772.      30560. : SKINTEMPMEAN  
   810 : 1994-12-30 00:00:00       0    18000       0 :      165.11  2.5544e+05  5.0564e+05 : SKINTEMPSTD   
   811 : 1994-12-30 00:00:00       0    18000       0 :     -16.912      2.6247      16.684 : U10MAX        
   812 : 1994-12-30 00:00:00       0    18000       0 :     -14.980      9.3093      21.272 : V10MAX        
   813 : 1994-12-30 00:00:00       0    18000       0 :      3.4161      12.964      21.708 : SPDUV10MAX    
   814 : 1994-12-30 00:00:00       0    18000       0 :  5.1991e+05  5.2232e+05  5.2272e+05 : TSPDUV10MAX   
   815 : 1994-12-30 00:00:00       0    18000       0 :     -11915.      5682.6      21167. : U10MEAN       
   816 : 1994-12-30 00:00:00       0    18000       0 :     -7008.3      9095.8      20802. : V10MEAN       
   817 : 1994-12-30 00:00:00       0    18000       0 :      3055.7      15339.      25922. : SPDUV10MEAN   
   818 : 1994-12-30 00:00:00       0    18000       0 :      386.60      62100.  2.7611e+05 : U10STD        
   819 : 1994-12-30 00:00:00       0    18000       0 :      1909.5  1.0111e+05  3.1275e+05 : V10STD        
   820 : 1994-12-30 00:00:00       0    18000       0 :      6710.4  1.6321e+05  4.0116e+05 : SPDUV10STD    
   821 : 1994-12-30 00:00:00       0    18000       0 :      0.0000      0.0000      0.0000 : RAINCVMAX     
   822 : 1994-12-30 00:00:00       0    18000       0 :      0.0000  0.00096011   0.0080501 : RAINNCVMAX    
   823 : 1994-12-30 00:00:00       0    18000       0 :      0.0000  4.2937e+05  5.2038e+05 : TRAINCVMAX    
   824 : 1994-12-30 00:00:00       0    18000       0 :      0.0000  5.0600e+05  5.2272e+05 : TRAINNCVMAX   
   825 : 1994-12-30 00:00:00       0    18000       0 :      0.0000      0.0000      0.0000 : RAINCVMEAN    
   826 : 1994-12-30 00:00:00       0    18000       0 :      0.0000     0.47396      2.9296 : RAINNCVMEAN   
   827 : 1994-12-30 00:00:00       0    18000       0 :      0.0000      0.0000      0.0000 : RAINCVSTD     
   828 : 1994-12-30 00:00:00       0    18000       0 :      0.0000  0.00040185   0.0077470 : RAINNCVSTD

**Comment by dudhia** [comment]
Why are there 3 values on each line?

**Comment by dmoreno11** [comment]
Because cdo outputs the mean, max and min for each field of the file

**Comment by dudhia** [comment]
This example does not have a bad T2MAX. I can maybe understand if the mean
is bad,
but need to see evidence of T2MAX being bad.

**Comment by dmoreno11** [comment]
T2MAX iand T2MIN are of 16 degrees k, which is too low

**Comment by dudhia** [comment]
I see.

**Comment by dudhia** [comment]
so the order in cdo is min, mean, max

**Comment by dmoreno11** [comment]
Yes

**Comment by weiwangncar** [comment]
@dmoreno11 You said that some years are ok. Are bad numbers come and go during your 20-year run? These max/min values are not just for a single wrfout file, they are evaluated during the times in between wrf output times. Also can you post the namelists related to adaptive time step?

**Comment by dudhia** [comment]
I think the bad values are only in wrfxtrm not wrfout. I would like to know
if it is repeatable.

**Comment by dmoreno11** [comment]
Yes, the bad values are only in wrfxtrm files. This was the first thing I checked.

The bad values appear in one single timestep in one year, but some years are 100% ok.

I think wrfxtrm files are always daily files. My history interval is 3h. Below are the adaptative timestep options:

use_adaptive_time_step              = .true., 
 step_to_output_time                 = .false., ! if true can originate dt=0 and messes stuff up
 target_cfl                          = 1.2, 1.2,
 max_step_increase_pct               = 5, 51,
 starting_time_step                  = -1, -1,
 max_time_step                       = -1, -1,
 min_time_step                       = -1, -1,
 adaptation_domain                   = 1,

**Comment by dmoreno11** [comment]
By the way, you guys should also take a look at the bug with step_to_output_time. 

If  step_to_output_time = .true.,, dt=0 can occur and crash the run . This happened to me several times

**Comment by dudhia** [comment]
We might have heard of this one with another user.
However, using false here may cause problems with wrfxtrm that needs to
also know the output times.
Are your output times then mostly not exact hours?

**Comment by dmoreno11** [comment]
I don't think so because my wrfouts and wrfxtrm files always end in 00:00:00. But they shouldn't be when I set to false... Weird

**Comment by dmoreno11** [comment]
I don't think so because my wrfouts and wrfxtrm files always end in 00:00:00. But they shouldn't be when I set to false... Weird

**Comment by weiwangncar** [comment]
@dmoreno11 This may not be the cause of the problem, but not setting the following:
max_time_step = -1, -1,
min_time_step = -1, -1,
isn't what we recommend. Setting bounds of the max and min time steps will avoid wild time step swings that we saw in our early tests, which is not good for model solutions.

**Comment by dmoreno11** [comment]
I followed the recommendations of ARW users guide on this...

**Comment by weiwangncar** [comment]
@dmoreno11 I believe this is what UG says:
max_time_step : use fixed values for all domains, e.g. 8*DX
min_time_step : use fixed values for all domains, e.g. 3*DX

**Comment by dmoreno11** [comment]
If you scroll down to description of namelist variables, you'll see that -1 implies max dt = 8dx and min of 3 dx

**Comment by weiwangncar** [comment]
@dmoreno11 You are right there. Do you still have an rsl file around that you can check if there are any time steps that go beyond the limits?

**Comment by weiwangncar** [comment]
@dmoreno11 I see now that time step bounds are set with -1 values. Thanks.

**Comment by dmoreno11** [comment]
Unfortunately no, the runs finished some time ago and I erased all rsl files... Anyway, if that would be the case cfl violations would likely occur and/or the run crash because I can't imagine a run not crashing with very large dt's.

Even more I think it's not the case because wrfout files are normal and this only happens for isolated times in years of integration. And some vars are ok

**Comment by dmoreno11** [comment]
Anyway, you guys are planning to take a look at this?

**Comment by dudhia** [comment]
Can you see if a restart repeats this problem exactly? Those types are
easier to help with.

**Comment by dmoreno11** [comment]
Also deleted the wrfrst files but since the first bad value appeared early on the simulation, I can run a  new one and see if it appears again

**Comment by dmoreno11** [comment]
Tried with a different domain (smaller), updated to WRF 4.2.2 and the exact same issue appears, at the same date

**Comment by dmoreno11** [comment]
and nothing wrong in the corresponding wrfout

**Comment by dmoreno11** [comment]
Minimum        Mean     Maximum : Parameter name                   
   793 : 0000-00-00 00:00:00       0     8400       0 :     -6.6020      6.5673      13.541 : T2MIN         
   794 : 0000-00-00 00:00:00       0     8400       0 :      3.1988      13.305      16.676 : T2MAX         
   795 : 0000-00-00 00:00:00       0     8400       0 :  5.2128e+05  5.2155e+05  5.2222e+05 : TT2MIN        
   796 : 0000-00-00 00:00:00       0     8400       0 :  5.2197e+05  5.2267e+05  5.2272e+05 : TT2MAX        
   797 : 0000-00-00 00:00:00       0     8400       0 :     -535.55      18095.      27228. : T2MEAN        
   798 : 0000-00-00 00:00:00       0     8400       0 :      3776.6  2.1078e+05  4.0501e+05 : T2STD         
   799 : 0000-00-00 00:00:00       0     8400       0 :   0.0025799   0.0053057   0.0078175 : Q2MIN         
   800 : 0000-00-00 00:00:00       0     8400       0 :   0.0058466   0.0091586    0.010272 : Q2MAX         
   801 : 0000-00-00 00:00:00       0     8400       0 :  5.2128e+05  5.2148e+05  5.2224e+05 : TQ2MIN        
   802 : 0000-00-00 00:00:00       0     8400       0 :  5.2221e+05  5.2271e+05  5.2272e+05 : TQ2MAX        
   803 : 0000-00-00 00:00:00       0     8400       0 :      7.5988      13.082      16.421 : Q2MEAN        
   804 : 0000-00-00 00:00:00       0     8400       0 :    0.031979    0.097813     0.14753 : Q2STD         
   805 : 0000-00-00 00:00:00       0     8400       0 :     -11.382      6.9258      15.836 : SKINTEMPMIN   
   806 : 0000-00-00 00:00:00       0     8400       0 :     0.37469      12.873      22.480 : SKINTEMPMAX   
   807 : 0000-00-00 00:00:00       0     8400       0 :  5.2128e+05  5.2151e+05  5.2222e+05 : TSKINTEMPMIN  
   808 : 0000-00-00 00:00:00       0     8400       0 :  5.2137e+05  5.2251e+05  5.2272e+05 : TSKINTEMPMAX  
   809 : 0000-00-00 00:00:00       0     8400       0 :     -2417.8      18749.      29431. : SKINTEMPMEAN  
   810 : 0000-00-00 00:00:00       0     8400       0 :      118.62  2.3347e+05  4.6898e+05 : SKINTEMPSTD   
   811 : 0000-00-00 00:00:00       0     8400       0 :     -8.7881      4.5282      12.755 : U10MAX        
   812 : 0000-00-00 00:00:00       0     8400       0 :     -11.449      10.941      21.030 : V10MAX        
   813 : 0000-00-00 00:00:00       0     8400       0 :      5.7089      12.624      21.097 : SPDUV10MAX    
   814 : 0000-00-00 00:00:00       0     8400       0 :  5.2128e+05  5.2240e+05  5.2272e+05 : TSPDUV10MAX   
   815 : 0000-00-00 00:00:00       0     8400       0 :     -2756.8      6773.7      15155. : U10MEAN       
   816 : 0000-00-00 00:00:00       0     8400       0 :      274.94      9875.3      18502. : V10MEAN       
   817 : 0000-00-00 00:00:00       0     8400       0 :      4468.7      14239.      25046. : SPDUV10MEAN   
   818 : 0000-00-00 00:00:00       0     8400       0 :      1101.5      44523.  1.3319e+05 : U10STD        
   819 : 0000-00-00 00:00:00       0     8400       0 :      9247.8  1.0035e+05  2.6937e+05 : V10STD        
   820 : 0000-00-00 00:00:00       0     8400       0 :      15581.  1.4487e+05  3.6621e+05 : SPDUV10STD    
   821 : 0000-00-00 00:00:00       0     8400       0 :      0.0000      0.0000      0.0000 : RAINCVMAX     
   822 : 0000-00-00 00:00:00       0     8400       0 :      0.0000   0.0010664   0.0057616 : RAINNCVMAX    
   823 : 0000-00-00 00:00:00       0     8400       0 :  5.2128e+05  5.2128e+05  5.2128e+05 : TRAINCVMAX    
   824 : 0000-00-00 00:00:00       0     8400       0 :  5.2128e+05  5.2209e+05  5.2272e+05 : TRAINNCVMAX   
   825 : 0000-00-00 00:00:00       0     8400       0 :      0.0000      0.0000      0.0000 : RAINCVMEAN    
   826 : 0000-00-00 00:00:00       0     8400       0 :      0.0000     0.51650      2.7770 : RAINNCVMEAN   
   827 : 0000-00-00 00:00:00       0     8400       0 :      0.0000      0.0000      0.0000 : RAINCVSTD     
   828 : 0000-00-00 00:00:00       0     8400       0 :      0.0000  0.00048706   0.0095322 : RAINNCVSTD

**Comment by dmoreno11** [comment]
also the times are wrong, even in 4.2.2

**Comment by lmarelle** [comment]
Hello all,

I ran into the same issue and found a fix for these bugs.
In a nutshell, there are 2 issues with adaptive time stepping in phys/module_diag_cl.F

1) Mean and standard deviation values are accumulated between two wrfxtrm output times, then divided by the number of time steps in the wrfxtrm output interval. This does not work for adaptive time step, since the time step changes, so the current approach overweights time periods when shorter time steps are used. To fix this, the average has to be weighted by the actual time steps used.

2) clwrf_output_calc checks if the current or next model time are wrfxtrm output times, but these checks are also broken for adaptive time steps, since the model checking if the number of time steps since restart, calculated as curr_secs2/dt, is a multiple of the the number of time steps in a wrfxtrm output frequency, clwrfH*60./dt. This does not work properly if the time step changes.

Here is the link to a commit fixing these issues:
https://github.com/lmarelle/wrf-france/commit/ded104713bacb567e387b81e52408ac9cd4ddf27

**Comment by smileMchen** [comment]
@dmoreno11 
Would you please upload your namelist.input for me to take a look? Thanks.

**Comment by dmoreno11** [comment]
> @dmoreno11 Would you please upload your namelist.input for me to take a look? Thanks.

 &time_control
 run_days                            = 0,
 run_hours                           = 0,
 run_minutes                         = 0,
 run_seconds                         = 0,
 start_year                          = 1995, 1995, 1994,
 start_month                         = 12,   12,   02,
 start_day                           = 01,   01,   01,
 start_hour                          = 00,   00,   00,
 start_minute                        = 00,   00,   00,
 start_second                        = 00,   00,   00,
 end_year                            = 1997, 1997, 2015,
 end_month                           = 01,   01,   01,
 end_day                             = 01,   01,   01,
 end_hour                            = 00,   00,   00,
 end_minute                          = 00,   00,   00,
 end_second                          = 00,   00,   00,
 interval_seconds                    = 21600
 input_from_file                     = .true.,.true.,.true.,
 history_interval                    = 360,  360,  180,
 frames_per_outfile                  = 124,  124,  248,
 restart                             = .false.,
 override_restart_timers             = .true.
 write_hist_at_0h_rst                = .true.
 restart_interval                    = 44640,
 io_form_history                     = 2,
 io_form_restart                     = 2,
 io_form_input                       = 2,
 io_form_boundary                    = 2,
 debug_level                         = 0,
 auxinput4_inname                    = "wrflowinp_d<domain>",
 auxinput4_interval                  = 360,   360,   360,
 io_form_auxinput4                   = 2,
 output_diagnostics                  = 1,
 auxhist3_interval                   = 1440,  1440,  1440,
 auxhist3_outname                    = "wrfxtrm_d<domain>_<date>",
 io_form_auxhist3                    = 2,
 frames_per_auxhist3                 = 31,   31,   31,
 /

 &domains
 time_step                           = 180,
 time_step_fract_num                 = 0,
 time_step_fract_den                 = 1,
 max_dom                             = 2,
 e_we                                = 65, 71,
 e_sn                                = 75, 121,
 e_vert                              = 33,  33,
 p_top_requested                     = 5000,
 num_metgrid_levels                  = 48,
 num_metgrid_soil_levels             = 5,
 dx                                  = 30000, 6000,
 dy                                  = 30000, 6000,
 grid_id                             = 1,     2,
 parent_id                           = 1,     1,
 i_parent_start                      = 1,     21,
 j_parent_start                      = 1,     33,
 parent_grid_ratio                   = 1,     5,
 parent_time_step_ratio              = 1,     5,
 feedback                            = 1,
 use_adaptive_time_step              = .true.,
 step_to_output_time                 = .false., ! if true can originate dt=0 and messes stuff up
 target_cfl                          = 1.2, 1.2,
 max_step_increase_pct               = 5, 51,
 starting_time_step                  = -1, -1,
 max_time_step                       = -1, -1,
 min_time_step                       = -1, -1,
 adaptation_domain                   = 1,
 /

 &physics
 mp_physics                          = 6,     6,     6,
 ra_lw_physics                       = 4,     4,     4,
 ra_sw_physics                       = 4,     4,     4,
 radt                                = 30,    10,    6,
 sf_sfclay_physics                   = 1,     1,     1,
 sf_surface_physics                  = 2,     2,     2,
 bl_pbl_physics                      = 1,     1,     1,
 bldt                                = 0,     0,     0,
 cu_physics                          = 3,     3,     0,
 cudt                                = 0,     0,     0,
 num_soil_layers                     = 4,
 num_land_cat                        = 21,
 sf_urban_physics                    = 0,     0,     0,
 sst_update                          = 1,
 usemonalb                           = .true.,
 surface_input_source                = 1,
 tmn_update                          = 1,
 lagday                              = 150,
 sst_skin                            = 1,
 bucket_mm                           = 100.0,
 bucket_J                            = 1.e9,
 seaice_threshold                    = 0.,
 fractional_seaice                   = 1,
 /

 &fdda
 grid_fdda                           = 2,      0,
 gfdda_inname                        = "wrffdda_d<domain>",
 gfdda_end_h                         = 999999,   0,
 gfdda_interval_m                    = 360,    0,
 fgdt                                = 0,      0,
 fgdtzero                            = 0,      0,
 if_no_pbl_nudging_uv                = 1,      1,
 if_no_pbl_nudging_t                 = 1,      1,
 if_no_pbl_nudging_ph                = 1,      1,
 if_zfac_uv                          = 1,      1,
  k_zfac_uv                          = 10,     10,
 if_zfac_t                           = 1,      1,
  k_zfac_t                           = 10,     10,
 if_zfac_ph                          = 1,      1,
  k_zfac_ph                          = 10,     10,
 dk_zfac_uv                          = 1,      1,
 dk_zfac_t                           = 1,      1,
 dk_zfac_ph                          = 1,      1,
 guv                                 = 0.0003, 0.0003,
 gt                                  = 0.0003, 0.0003,
 gph                                 = 0.0003, 0.0003,
 xwavenum                            = 2,
 ywavenum                            = 2,
 if_ramping                          = 0,
 dtramp_min                          = 60.0,
 grid_sfdda                          = 0,
 io_form_gfdda                       = 2,
 /

 &dynamics
 w_damping                           = 1,
 diff_opt                            = 1,
 km_opt                              = 4,
 diff_6th_opt                        = 0,      0,      0,
 diff_6th_factor                     = 0.12,   0.12,   0.12,
 base_temp                           = 290.
 damp_opt                            = 3,
 zdamp                               = 5000., 5000., 5000.,
 dampcoef                            = 0.2,    0.2,    0.2,
 khdif                               = 0.,     0.,     0.,
 kvdif                               = 0.,     0.,     0.,
 non_hydrostatic                     = .true.,.true.,.true.,
 moist_adv_opt                       = 1,      1,    1,
 scalar_adv_opt                      = 1,      1,    1,
 /
! epssm                               = 0.3,    0.5,

 &bdy_control
 spec_bdy_width                      = 10,
 spec_zone                           = 1,
 relax_zone                          = 9,
 spec_exp                            = 0.33,
 specified                           = .true.,.false.,.false.,
 nested                              = .false.,.true.,.true.,
 /

 &grib2
 /

 &namelist_quilt
 nio_tasks_per_group = 0,
 nio_groups = 1,
 /

**Comment by lmarelle** [comment]
Concerning my last message, I have been running more tests and there are still issues with this condition:
`IF (MOD(NINT(curr_secs2+dt),NINT(clwrfH*60.)) == 0) THEN`
It is not always valid at the last time step before writing the wrfxtrm. I suppose the +dt could cause issues if the last time step is different from the second-to-last time step, if the adaptive time step routine is called after the climate diagnostics routine. I am investigating.

**Comment by LluisFB** [comment]
Hi, I am the one who wrote the first version of the whole `wrfxtrm`. 

At that time, I was not that proficient in WRF, thus I came out with this not suitable for all cases solution of counting time-steps (I apologize for the issues). Maybe it would be worth to use a more reliable robust way, through the use of the current time value of the integration. Something like (calcualted at the `module_diagnostics_driver.F`):
`is_output_t = ( Currtime .ge. aux3_time + aux3int - dtint )` 
I do not know if this semantic works in adaptative time step

And then passe `is_output_t` into the subroutine `clwrf_output_calc` and use it to finish the computation of statistics. Maybe it would be also worth to add a new variable in Registry (if it is not still there already) to account for the amount of time steps?
